spec/cancan/model_adapters/active_record_adapter_spec.rb

```diff
@@ -909,6 +909,37 @@ class JsonTransaction < ActiveRecord::Base
     end
   end
 
+  context 'with rule application to subclass for non sti class' do
+    before do
+      ActiveRecord::Schema.define do
+        create_table :parents, force: true do |t|
+        end
+
+        create_table :children, force: true do |t|
+        end
+      end
+
+      class ApplicationRecord < ActiveRecord::Base
+        self.abstract_class = true
+      end
+
+      class Parent < ActiveRecord::Base
+      end
+
+      class Child < Parent
+      end
+    end
+
+    it 'rules are not effecting parent class' do
```

Comment:
I think we also need the test where can'ing the child doesn't automatically can the parent to satisfy this scenario, yeah? https://github.com/CanCanCommunity/cancancan/issues/677#issuecomment-753387317

Something like...

```
u1 = User.create!(name: 'pippo')
ability = Ability.new(u1)
ability.can :manage, Child
expect(ability).to be_able_to(:index, Child)
expect(ability).not_to be_able_to(:index, Parent)
```